services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -13,38 +13,89 @@ import scala.concurrent.{ ExecutionContext, Future }
 class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
     extends ProjectVersioning[VersioningException] {
 
-  override def updateFile(project: Project, projectFile: ProjectFile, version: Option[Version])(
+  private def getLastProjectVersion(project: Project)(implicit ec: ExecutionContext): Future[Option[PipelineVersion]] =
+    getProjectVersions(project).map {
+      case Left(error)                      => throw error
```

Comment:
Throwing an exception is not the Scala way, you should avoid it wherever possible.
It is better to return `Future.failed` instead.

Reply:
Yep, will change it